client.go

```diff
@@ -88,6 +95,30 @@ type StreamOptions struct {
 	// this will behave as a stream with a single partition. If this is not
 	// set, it defaults to 1.
 	Partitions int32
+
+	// The maximum size a stream's log can grow to, in bytes, before we will discard
+	// old log segments to free up space. A value of 0 indicates no limit.
+	RetentionMaxBytes int64
+	// The maximum size a stream's log can grow to, in number of messages, before we will discard
+	// old log segments to free up space. A value of 0 indicates no limit.
+	RetentionMaxMessages int64
+	// The TTL for stream log segment files, after which they are deleted. A value of 0 indicates no TTL.
+	RetentionMaxAge int64
```

Comment:
We should use `time.Duration` and handle converting it to the int64. Using `time.Duration` provides a nicer API experience to the end user IMO.

Reply:
Agree. It's more natural to configure in `time.Duration`. the API should handle the conversion behind the scene